aeron-cluster/src/main/java/io/aeron/cluster/ConsensusModuleAgent.java

```diff
@@ -1771,14 +1771,10 @@ int catchupPoll(final long limitPosition, final long nowNs)
                 throw new ClusterEvent(
                     "unexpected image close during catchup: position=" + logAdapter.image().position());
             }
-        }
 
-        final long appendPosition = logAdapter.position();
-        if (appendPosition > lastAppendPosition || nowNs > (timeOfLastAppendPositionSendNs + leaderHeartbeatIntervalNs))
-        {
-            commitPosition.proposeMaxOrdered(appendPosition);
             final ExclusivePublication publication = election.leader().publication();
-            tryUpdateAppendPosition(publication, nowNs, replayLeadershipTermId, appendPosition);
+            tryUpdateAppendPosition(publication, nowNs, replayLeadershipTermId, appendPosition.get());
```

Comment:
If no progress then we should only pulse out append position on an interval otherwise we could fill the window and prevent other messages getting out in a timely manner which may be more important.